### PR TITLE
EES-1595 - corrected early termination of the import recovery process…

### PR DESCRIPTION
**Pull request by duncan-at-hiveit**
… in the event that there was a file that needed batching but no batches had yet been created.

The run.js file is included as a change in this PR just because I amended its file permissions to be executable.

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Data.Processor/ImportRecoveryHandler.cs [hunk 1]
```diff
@@ -66,19 +66,24 @@ public static void CheckIncompleteImports(string connectionString)
...
                         var batches = GetBatchesRemaining(blobContainer, entity.PartitionKey, m.OrigDataFileName)
                             .Result;
 
                         // If no batches then assume it didn't get passed initial validation stage
                         if (!batches.Any())
                         {
+                            Console.WriteLine($"No batches created for {entity.PartitionKey} : {entity.RowKey}`` although batches are necessary to create.  Adding message to the queue to try re-import.");
                             pendingQueue.AddMessage(new CloudQueueMessage(entity.Message));
-                            return;
+                            continue;
```
Comment (ntsim) [inline src/GovUk.Education.ExploreEducationStatistics.Data.Processor/ImportRecoveryHandler.cs:79]: Ah the empty queues make a lot of sense now!

Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Data.Processor/ImportRecoveryHandler.cs:79]: :)

File: src/GovUk.Education.ExploreEducationStatistics.Data.Processor/ImportRecoveryHandler.cs [hunk 2]
```diff
@@ -66,19 +66,24 @@ public static void CheckIncompleteImports(string connectionString)
...
                     {
+                        Console.WriteLine($"Creating messages for remaining batches for {entity.PartitionKey} : {entity.RowKey}");
                         var m = JsonConvert.DeserializeObject<ImportMessage>(entity.Message);
                         var batches = GetBatchesRemaining(blobContainer, entity.PartitionKey, m.OrigDataFileName)
                             .Result;
 
                         // If no batches then assume it didn't get passed initial validation stage
                         if (!batches.Any())
                         {
+                            Console.WriteLine($"No batches created for {entity.PartitionKey} : {entity.RowKey}`` although batches are necessary to create.  Adding message to the queue to try re-import.");
```
Comment (ntsim) [inline src/GovUk.Education.ExploreEducationStatistics.Data.Processor/ImportRecoveryHandler.cs:77]: Random backticks?

**Comment by ntsim** [comment]
Would recommend squashing your commits as well as it looks like the same one has been duplicated?

File: src/GovUk.Education.ExploreEducationStatistics.Data.Processor/ImportRecoveryHandler.cs (hunk 2 above)
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Data.Processor/ImportRecoveryHandler.cs]: Random backticks removed, thanks :)

**Review by ntsim — APPROVED** [review]

**Pull request merged**